hack/clean-deploy.sh

```diff
@@ -6,7 +6,7 @@ OC_TOOL="${OC_TOOL:-oc}"
 profiles=$(${OC_TOOL} get performanceprofile -o name)
 for profileName in $profiles
 do
-  nodeSelector="$(${OC_TOOL} get $profileName -o=jsonpath='{.spec.nodeSelector}'  | awk -F'[/:]' '{print $2}')"
+  nodeSelector="$(${OC_TOOL} get $profileName -o=jsonpath='{.spec.nodeSelector}'  | awk -F'[/:"]' '{print $3}')"
```

Comment:
Can we use this instead? awk and print seem to be unreliable. 

`basename `oc get performanceprofiles.performance.openshift.io performance -o json | jq -cr '.spec.nodeSelector | keys[]'``

Reply:
+1. If we can without them... I am not referring to the proposed change. In general.

Reply:
I like more the jq approach, but could you please clarify what (and when?) awk is unreliable?

Reply:
Like in this case where the required data was not returned. The awk in itself is not unreliable though. With jq approach, we don't have to worry about the delimiter and print field.

Reply:
thanks, good point indeed. Definitely +1 for jq.

Reply:
+1 on using jq.
that's also what we're doing in cnf-feature-deploy and d/s

Reply:
Just my 5c, IIRC jq not installed by default under the OS, and awk installed.

Reply:
Right. So if we use jq, we need to add a check jq is there before we actually run commands that can modify the cluster state.

Reply:
let me explain slightly. First of all I agree with @cynepco3hahue that **jq** may not be present on a server (not a big deal, of course, we can add verification).

Secondly - even if we use jq command - we will need to split the result:

```
# oc get performanceprofiles.performance.openshift.io manual -o json | jq -cr '.spec.nodeSelector | keys[]'
node-role.kubernetes.io/worker-cnf
```
From this output I need to get only **_worker-cnf_**, so anyway I need to use **awk** or something similar for splitting it on 2 parts.. so, the command in this case will look like:
 
`oc get performanceprofiles.performance.openshift.io manual -o json | jq -cr '.spec.nodeSelector | keys[]' | awk -F "/" '{print $2}'`


Comparing this two commands - I don't see any benefits from switching to jq..

Reply:
You can use `basename node-role.kubernetes.io/worker-cnf` instead of awk.

Reply:
@dshchedr while I see @gsr-shanks 's point, I've nothing against awk, so considering your last point I'm fine keeping awk.